### PR TITLE
Align job history with tide history view

### PR DESCRIPTION
**Pull request by YuviGold**
- Move formatDuration to common package

Remove `parseDuration` as it isn't used.
Align `cell.time` tooltip with `prRevision`.

- Move job history builds rendering to typescript module

Add a `Revision` column and align with tide history and deck PR revision view.
`Started` column time display is aligned with the tide history view.
Now it would show a tooltip when hovering over it, and use the same TZ
Support multiple pulls for batch jobs.

Screenshot from 2021-07-01 20-15-03

### CODE REVIEW

**Comment by chaodaiG** [comment]
Thank you!

a quick question before going over the code change: the time displayed on prow.k8s.io is PST timezone, while job history is UTC, can we make them the same to be consistent by the way?

File: prow/cmd/deck/job_history.go [hunk 1]
```diff
@@ -371,7 +371,7 @@ func getBuildData(ctx context.Context, bucket storageBucket, dir string) (buildD
 		logrus.WithError(err).Debugf("failed to read %s", prowv1.ProwJobFile)
 	} else {
 		if pj.Spec.Refs != nil && len(pj.Spec.Refs.Pulls) > 0 {
```
Comment (chaodaiG) [inline prow/cmd/deck/job_history.go:373]: do we still need `len(pj.Spec.Refs.Pulls) > 0 `?

File: prow/cmd/deck/job_history.go [hunk 2]
```diff
@@ -67,7 +67,7 @@ type buildData struct {
 	Duration     time.Duration
 	Result       string
 	commitHash   string
-	Pull         *prowv1.Pull
+	Pulls        []prowv1.Pull
```
Comment (chaodaiG) [inline prow/cmd/deck/job_history.go:70]: wouldn't this break `.Pull` from https://github.com/kubernetes/test-infra/pull/22747/files?

**Review by chaodaiG — CHANGES_REQUESTED** [review]

File: prow/cmd/deck/job_history.go (hunk 2 above)
Comment (YuviGold) [reply prow/cmd/deck/job_history.go:70]: The code was changed accordingly 
https://github.com/kubernetes/test-infra/pull/22760/files#diff-5f8d6554e048de4637c1ba0e5ba42031d49e9dff2021510486955ebe338054b0R27-R33

**Comment by YuviGold** [comment]
> Thank you!
> 
> a quick question before going over the code change: the time displayed on prow.k8s.io is PST timezone, while job history is UTC, can we make them the same to be consistent by the way?

Actually, I didn't mention that as part of the commit message since I looked only for the tooltip preview, so I used the same function. But it was resolved as part of it! :)

File: prow/cmd/deck/job_history.go (hunk 1 above)
Comment (YuviGold) [reply prow/cmd/deck/job_history.go:373]: I believe you are right. Will fix.

**Review by chaodaiG — APPROVED** [review]

**Pull request merged**